CMakeLists.txt

```diff
@@ -2,6 +2,10 @@ cmake_minimum_required(VERSION 3.14)
 
 project(ClickHouse LANGUAGES C CXX ASM)
 
+# If a ClickHouse CMake option without "CLICKHOUSE_" prefix is specified, issue a warning and set the corresponding option with prefix to
+# the specified value. To be removed after a grace period of three months (Oct 22)
+include (cmake/set_prefixed_option_if_needed.cmake)
```

Comment:
Are you sure about preserving the compatibility, especially for all contrib modules?
This may indeed improve the translation period for developers, although I'm not sure since not all look at cmake warnings, while the complexity of cmake rules slightly increased.

Reply:
I for my part have some aliases in my .bashrc which invoke CMake with custom CMake variables. I could imagine that's common practice, e.g., to do slimmed down / full debug/  release builds etc.

On the other hand, CMake complains if passed variables are not used:

```
CMake Warning:
  Manually-specified variables were not used by the project:

    ENABLE_AVRO
```

Plus, I suppose that the majority of variables obscure enough that developers don't pass them anyways. The "common" ones may just be ENABLE_LIBRARIES, ENABLE_EMBEDDED_COMPILER, USE_STATIC_LIBRARIES and SPLIT_SHARED_LIBRARIES.

So, weighting all that with the complexity of the compat handling, you may be right and we could do a "big bang" transition instead...